server/src/main/java/org/opensearch/plugins/SearchPlugin.java

```diff
@@ -178,6 +184,36 @@
         return emptyList();
     }
 
+    /**
```

Comment:
@nknize @andrross guys would really appreciate the feedback if I am on the right track here, moving towards sandboxing concurrent search implementation (https://github.com/opensearch-project/OpenSearch/pull/1500) . 

Basically, the idea  is to provide the extension points to allow plugging in different searcher implementation(s) at `QueryPhase` time (the default would be the current sequential scan). In this case the core would know nothing about concurrent / non-concurrent nature of the search over Apache Lucene segments. But the sandbox plugin, if installed, could tweak that (even on per index basis) and substitute the searcher with concurrent one. 

Thank you.

Reply:
Long-term, is it possible for concurrent search to become the only implementation, even if there are some cases where it is configured to run with a single thread?

Reply:
I think so, yeah, there are two flows to address (aggregations and forced early termination) but I don't see the reasons why it couldn't become the only implementation in future.

Reply:
My initial thought is that it would be good to get towards a single implementation for something like this and so it might not be worth it to make it pluggable at this point. Would love to get Nick's @nknize opinion on this as well.

Reply:
Yeah, the initial implementation (https://github.com/opensearch-project/OpenSearch/pull/1500) is doing exactly that, but @nknize had the concerns and suggested to sandbox the changes.

Reply:
lol, then let's definitely get Nick's input on this!